mod_reforged/hooks/skills/special/double_grip.nut

```diff
@@ -0,0 +1,307 @@
+::Reforged.HooksMod.hook("scripts/skills/special/double_grip", function(q) {
+	// Is set during onUpdate so that hybrid weapons are properly dealt with in functions other than onUpdate
+	// because we go through weapon types alphabetically and choose the first applicable type
+	q.m.CurrWeaponType <- null;
+
+	q.create = @(__original) function()
+	{
+		__original();
+		this.m.Description = "With the second hand free, this character can adopt a more versatile fighting stile or get a firm double grip on his weapon to inflict additional damage."
+	}
+
+	q.canDoubleGrip = @() function()
+	{
+		local actor = this.getContainer().getActor();
+		local weapon = actor.getMainhandItem();
+		if (weapon == null || !weapon.isDoubleGrippable())
+			return false;
+
+		local offhand = actor.getOffhandItem();
+		if (offhand == null)
+			return true;
+
+		return offhand.getStaminaModifier() < 10 && this.m.CurrWeaponType == "SouthernSword" && this.getContainer().hasSkill("perk.rf_en_garde");
+	}
+
+	q.applyBonusOnUpdate <- function( _properties )
+	{
+		switch (this.m.CurrWeaponType)
+		{
+			case ::Const.Items.WeaponType.Axe:
+				_properties.MeleeDamageMult *= 1.15;
```

Comment:
All of these values should be made into member variables.

Reply:
That is too much in this case imo as there's too many values. I instead implemented these separate functions with a switch statement so submodders who want to modify a particular weapon's behavior can hook these functions to easily modify the behavior of a particular weapon.